src/web/source.rs

```diff
@@ -101,48 +103,54 @@ impl FileList {
+                        let path = file[1].as_string().unwrap();
+
+                        // skip .cargo-ok generated by cargo
+                        if path == ".cargo-ok" {
+                            return None;
+                        }
 
-                    // look only files for req_path
-                    if path.starts_with(&req_path) {
-                        // remove req_path from path to reach files in this directory
-                        let path = path.replace(&req_path, "");
-                        let path_splited: Vec<&str> = path.split('/').collect();
-
-                        // if path have '/' it is a directory
-                        let ftype = if path_splited.len() > 1 {
-                            FileType::Dir
-                        } else if mime.starts_with("text") && path_splited[0].ends_with(".rs") {
-                            FileType::RustSource
-                        } else if mime.starts_with("text") {
-                            FileType::Text
-                        } else {
-                            FileType::Binary
-                        };
-
-                        let file = File {
-                            name: path_splited[0].to_owned(),
-                            file_type: ftype,
-                        };
-
-                        // avoid adding duplicates, a directory may occur more than once
-                        if !file_list.contains(&file) {
-                            file_list.push(file);
+                        // look only files for req_path
+                        if path.starts_with(&req_path) {
+                            // remove req_path from path to reach files in this directory
+                            let path = path.replace(&req_path, "");
+                            let path_splited: Vec<&str> = path.split('/').collect();
+
+                            // if path have '/' it is a directory
+                            let ftype = if path_splited.len() > 1 {
+                                FileType::Dir
+                            } else if mime.starts_with("text") && path_splited[0].ends_with(".rs") {
+                                FileType::RustSource
+                            } else if mime.starts_with("text") {
+                                FileType::Text
+                            } else {
+                                FileType::Binary
+                            };
+
+                            let file = File {
+                                name: path_splited[0].to_owned(),
+                                file_type: ftype,
+                            };
+
+                            // avoid adding duplicates, a directory may occur more than once
+                            return Some(file);
                         }
                     }
-                };
-            }
+
+                    None
+                })
+                .collect::<Vec<_>>();
+
+            file_list.dedup();
```

Comment:
This does not preserve the same behavior. `dedup` only works if the array is already sorted, and these files may not be. Also, the `// avoid duplicates` comment should be here instead of above.

Reply:
My bad, that's a rather counter-intuitive implementation. Would you prefer me to sort or to revert to the previous implementation?

Reply:
Let's revert, the PR is already big enough.

Reply:
@Kixiron can you revert this change please?